docs/EXTENSIBILITY.md

```diff
@@ -0,0 +1,83 @@
+# Spice.ai Extensibility
+
+This document is an overview of all the interfaces and extension points in Spice.ai.
+
+## DataSource
```

Comment:
Maybe we should continue to call these `DataConnectors` especially if they now include writes.